src/main.c

```diff
@@ -207,53 +272,115 @@ int main(int argc, char **argv)
-			if(unit_test_storage()) exit(1);
-			fprintf(stderr, "\n\nALL TESTS PASSED\n\n");
-			exit(0);
+			else if(strcmp(argv[i], "-ch") == 0 && (i+1) < argc) {
+				config_set("global", "host access prefix", argv[i+1]);
+				fprintf(stderr, "%s: deprecate option -- %s -- please use -s instead.\n", argv[0], argv[i]);
+				remove_option(i, &argc, argv);
+			}
+			else i++;
 		}
-		else {
-			fprintf(stderr, "Cannot understand option '%s'.\n", argv[i]);
-			fprintf(stderr, "\nUSAGE: %s [-d] [-l LINES_TO_SAVE] [-u UPDATE_TIMER] [-p LISTEN_PORT] [-df debug flags].\n\n", argv[0]);
-			fprintf(stderr, "  -c CONFIG FILE the configuration file to load. Default: %s.\n", CONFIG_DIR "/" CONFIG_FILENAME);
-			fprintf(stderr, "  -l LINES_TO_SAVE can be from 5 to %d lines in JSON data. Default: %d.\n", RRD_HISTORY_ENTRIES_MAX, RRD_DEFAULT_HISTORY_ENTRIES);
-			fprintf(stderr, "  -t UPDATE_TIMER can be from 1 to %d seconds. Default: %d.\n", UPDATE_EVERY_MAX, UPDATE_EVERY);
-			fprintf(stderr, "  -p LISTEN_PORT can be from 1 to %d. Default: %d.\n", 65535, LISTEN_PORT);
-			fprintf(stderr, "  -u USERNAME can be any system username to run as. Default: none.\n");
-			fprintf(stderr, "  -ch path to access host /proc and /sys when running in a container. Default: empty.\n");
-			fprintf(stderr, "  -nd or -nodeamon to disable forking in the background. Default: unset.\n");
-			fprintf(stderr, "  -df FLAGS debug options. Default: 0x%08llx.\n", debug_flags);
-			fprintf(stderr, "  -stacksize BYTES to overwrite the pthread stack size.\n");
-			fprintf(stderr, "  -pidfile FILENAME to save a pid while running.\n");
-			exit(1);
+	}
+
+	// parse options
+	{
+		int num_opts = sizeof(options) / sizeof(struct option_def);
+		char optstring[(num_opts * 2) + 1];
+
+		int string_i = 0;
+		for( i = 0; i < num_opts; i++ ) {
+			optstring[string_i] = options[i].val;
+			string_i++;
+			if(options[i].arg_name) {
+				optstring[string_i] = ':';
+				string_i++;
+			}
 		}
+
+		int opt;
+		while( (opt = getopt(argc, argv, optstring)) != -1 ) {
+			switch(opt) {
+				case 'c':
+					if(load_config(optarg, 1) != 1) {
+						error("Cannot load configuration file %s.", optarg);
+						exit(1);
+					}
+					else {
+						debug(D_OPTIONS, "Configuration loaded from %s.", optarg);
+						config_loaded = 1;
+					}
+					break;
+				case 'D':
+					dont_fork = 1;
+					break;
+				case 'h':
+					help(0);
+					break;
+				case 'P':
+					strncpy(pidfile, optarg, FILENAME_MAX);
+					pidfile[FILENAME_MAX] = '\0';
+					break;
+				case 'p':
+					config_set("global", "port", optarg);
+					break;
+				case 's':
+					config_set("global", "host access prefix", optarg);
+					break;
+				case 't':
+					config_set("global", "update every", optarg);
+					break;
+				case 'u':
+					config_set("global", "run as user", optarg);
+					break;
+				case 'v':
+					// TODO: Outsource version to makefile which can compute version from git.
+					printf("netdata 1.1.0\n");
```

Comment:
Of course. My intention was not to do this in this pull request. Of course if it is OK for you.
I want to provide another pull request with the following enhancement.
- Enhance version check.
- Make -v portable.
- Compute version from git.

If you do not care this is hard-coded for some time I will leave this in.